test/robustness/overlay/areal_areal/random_integer_grids.cpp

```diff
@@ -0,0 +1,267 @@
+        const auto geo2 = bits_to_geometry(bits2, grid, points, settings, all_success);
+        test_op(bits1, bits2, geo1, geo2, "union", std::bit_or<>{},
+                [](mp const& g1, mp const& g2) { mp g; bg::union_(g1, g2, g); return g; },
+                points, grid, settings, all_success);
+        test_op(bits1, bits2, geo1, geo2, "intersection", std::bit_and<>{},
+                [](mp const& g1, mp const& g2) { mp g; bg::intersection(g1, g2, g); return g; },
+                points, grid, settings, all_success);
+        test_op(bits1, bits2, geo1, geo2, "sym_difference", std::bit_xor<>{},
+                [](mp const& g1, mp const& g2) { mp g; bg::sym_difference(g1, g2, g); return g; },
+                points, grid, settings, all_success);
+        test_op(bits1, bits2, geo1, geo2, "difference g1 \\ g2", 
+                [](std::bitset<chunk_size> b1, std::bitset<chunk_size> b2) { return b1 & (~b2); },
+                [](mp const& g1, mp const& g2) { mp g; bg::difference(g1, g2, g); return g; },
+                points, grid, settings, all_success);
+        test_op(bits1, bits2, geo1, geo2, "difference g2 \\ g1", 
+                [](std::bitset<chunk_size> b1, std::bitset<chunk_size> b2) { return b2 & (~b1); },
+                [](mp const& g1, mp const& g2) { mp g; bg::difference(g2, g1, g); return g; },
+                points, grid, settings, all_success);
+    }
+    return all_success;
+}
+
+int main(int argc, char** argv)
+{
+    BoostGeometryWriteTestConfiguration();
+    try
+    {
+        namespace po = boost::program_options;
+        po::options_description description("=== random_integer_grids ===\nAllowed options");
+
+        grid_settings settings;
+
+        description.add_options()
+            ("help", "Help message")
+            ("seed",
+             po::value<decltype(settings.seed)>(&settings.seed)->default_value(settings.seed),
+             "Initialization seed for random generator")
+            ("count",
+             po::value<decltype(settings.count)>(&settings.count)->default_value(settings.count),
+             "Number of tests (-1 for infinite loop)")
+            ("width",
+             po::value<decltype(settings.width)>(&settings.width)->default_value(settings.width),
+             "Width of grid")
+            ("height",
+             po::value<decltype(settings.height)>(&settings.height)->default_value(settings.height),
+             "Height of grid")
+        ;
+
+        po::variables_map varmap;
+        po::store(po::parse_command_line(argc, argv, description), varmap);
+        po::notify(varmap);
+
+        if (varmap.count("help"))
+        {
+            std::cout << description << std::endl;
+            return 1;
+        }
+        if ( ! test_all(settings) )
+            return 1;
+    }
+    catch(std::exception const& e)
+    {
+        std::cout << "Exception " << e.what() << '\n';
+    }
+    catch(...)
+    {
+        std::cout << "Other exception" << '\n';
+    }
```

Comment:
I would like to see a summary of the number of tests, the number of failures, and the time spent.